### PR TITLE
feat(catalog): Add a title to the PaginatedCatalogTable.

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

Thread in discord [here](https://discord.com/channels/687207715902193673/923144214580191282/1186802058830024865).

This PR copies the title from `CatalogTable` to the `PaginatedCatalogTable`. The count without additional information may be confusing re: current page, but I think it's useful so I can decide whether to filter my search further. 

Screenshot 2024-01-25 at 7 13 29 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Approving for changesets etc. And looks good to me! @backstage/catalog-maintainers for final approval.

Possibly `count` vs `totalItems` ... since that's what it's called in the actual payload after all.

**Comment by drodil** [comment]
What happened @sennyeya ?

File: .changeset/perfect-chefs-act.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-react': minor
+---
+
+Number of results is now directly added as the field `count` on `useEntityList`.
```
[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply .changeset/perfect-chefs-act.md:5]: updated!

**Comment by aramissennyeydd** [comment]
This should be ready for a rereview, lighthouse tests seem to be failing across the project (#23344)

**Review by alde — APPROVED** [review]
LGTM - @backstage/maintainers may have more insight about the failing lighthouse tests

**Pull request merged**